File: src/main/java/org/springframework/samples/petclinic/service/interfaces/PetService.java
package org.springframework.samples.petclinic.service.interfaces;

import org.springframework.samples.petclinic.model.PetType;
import org.springframework.samples.petclinic.dtos.PetDTO;
import org.springframework.samples.petclinic.dtos.ResponseData;

import java.util.Collection;

public interface PetService {
    ResponseData<String> savePet(PetDTO pet);

    ResponseData<PetDTO> getPetById(int petId);

    ResponseData<Collection<PetDTO>> getPetsByOwnerId(int ownerId);

    ResponseData<Collection<PetType>> getPetTypes();
}

File: client/src/service/pet/petService.js
import clinicRequest from '../clinicRequest';

const endpoint = '/pet';

const getPetsbyOwnerId = ownerId => {
  return clinicRequest.get(endpoint + `/owner/${ownerId}`);
};

const savePet = pet => {
  return clinicRequest.post(endpoint, pet);
};

const getPetTypes = () => {
  return clinicRequest.get(endpoint + '/types');
};

const getPetById = petId => {
  return clinicRequest.get(endpoint + `/${petId}`);
};

const petService = { getPetsbyOwnerId, getPetTypes, savePet, getPetById };
export default petService;
